rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -73,6 +73,39 @@ impl Tun {
                 }
             };
 
+        let iface_idx = adapter.get_adapter_index()?;
+
+        // Remove any routes that were previously associated with us
+        // TODO: Pick a more elegant way to do this
+        Command::new("powershell")
+            .creation_flags(CREATE_NO_WINDOW)
+            .arg("-Command")
+            .arg(format!(
+                "Remove-NetRoute -InterfaceIndex {iface_idx} -Confirm:$false"
+            ))
+            .stdout(Stdio::null())
+            .status()?;
+
+        set_iface_config(adapter.get_luid(), DEFAULT_MTU)?;
+
+        let session = Arc::new(adapter.start_session(wintun::MAX_RING_CAPACITY)?);
+
+        let (packet_tx, packet_rx) = mpsc::channel(5);
+
+        let recv_thread = start_recv_thread(packet_tx, Arc::clone(&session))?;
+        let packet_rx = std::sync::Mutex::new(packet_rx);
```

Comment:
This Mutex can now be removed because we have `&mut self`.

Reply:
Can be done in a follow-up though of course.

Reply:
will do